static/app/components/charts/utils.tsx

```diff
@@ -200,9 +201,20 @@ export class GranularityLadder {
   }
 
   getInterval(minutes: number): string {
+    if (minutes < 0) {
+      // Sometimes this happens, in unknown circumstances. See the `getIntervalForMetricFunction` function span in Sentry for more info, the reason might appear there. For now, a reasonable fallback in these rare cases is to return the finest granularity, since it'll either fulfill the request or time out.
+      Sentry.withScope(() => {
+        Sentry.captureException(
```

Comment:
you don't need the withScope here, `Sentry.captureException` alone is fine, `withScope` only needed if you want to isolate scope changes (also call `Sentry.setTag`, `Sentry.setUser` but only affect the `withScope` callback)

`getInterval` have a side effect, so it has to be wrapped in `useEffect`. Do we need to document this?